File: src/test/java/com/bbcnews/automation/base/CharlesProxy.java
package com.bbcnews.automation.base;

import java.io.IOException;

public class CharlesProxy {

    static String charlesconfig = "/charles/CharlesConfig.xml";
    static String charlesRecord = "curl -v -x http://localhost:8888 http://control.charles/recording/start";
    static String charlesSessionDownload = "curl -o session.chls -x http://localhost:8888 http://control.charles/session/download";
    static String charlesStop = "curl -v -x http://localhost:8888 http://control.charles/recording/stop";
    static String startCharles = "Charles -config";
    static String stopCharles = "killall Charles";

    Runtime rt = Runtime.getRuntime();


    public void startCharles() {

        try {
            //Process pr = rt.exec("Charles -config /Users/ramakh02/Desktop/AppiumBBCNewsAndroid/charles/CharlesConfig.xml");
            Process pr = rt.exec("Charles");
        } catch (IOException e) {

        }
    }


    public void stopCharles() {
        try {
            Process pr1 = rt.exec("killall Charles");
        } catch (IOException e) {

        }
    }

    public void converttoCSV() {
        try {
            Process pr6 = rt.exec("Charles convert session.chls session.csv");
            Thread.sleep(5000);
        } catch (IOException | InterruptedException e) {
        }

    }

    public void downloadCharlesSession() throws InterruptedException {
        try {
            Process pr4 = rt.exec("curl -o session.chls -x http://localhost:8888 http://control.charles/session/download");
            Thread.sleep(5000);
        } catch (IOException e) {
        }
    }

    public void charlesSessionStart() {
        try {
            Process pr2 = rt.exec("curl -v -x http://localhost:8888 http://control.charles/recording/start");
            Thread.sleep(5000);
        } catch (Exception e) {
        }
    }

    public void stopCharlesSession() {
        try {
            //Process p5 = Runtime.getRuntime().exec(charlesStop);
            Process pr5 = rt.exec("curl -v -x http://localhost:8888 http://control.charles/recording/stop");
        } catch (IOException e) {
        }

    }

}

File: src/test/java/com/bbcnews/automation/testutils/CharlesProxy.kt
package com.bbcnews.automation.testutils

import com.bbcnews.automation.commonfunctions.CommonFunctionKotlin
import org.testng.Assert.assertEquals
import java.io.*

class CharlesProxy {

    var charlesconfig = "./charles/BBCNews.config"
    var charlesRecord = "curl -v -x http://localhost:8888 http://control.charles/recording/start"
    var charlesSessionDownload = "curl -o session.chls -x http://localhost:8888 http://control.charles/session/download"
    var charlesStop = "curl -v -x http://localhost:8888 http://control.charles/recording/stop"
    var startCharles = "Charles -config"
    var stopCharles = "killall Charles"

    var rt = Runtime.getRuntime()

    private var commonFunctionKotlin = CommonFunctionKotlin()

    var charlesFolder = commonFunctionKotlin.extentResultFolder("CharlesFolder")
    var files = File(charlesFolder)


    fun startCharles() {
        //val dir = File("./charles/BBCNews.config")
        try {
            // Process pr = rt.exec("charles");
            rt.exec("charles -config ./charles/charlesproxy.config &")
            System.out.println("Starting Charles ")
        } catch (e: IOException) {

        }

    }


    fun stopCharles() {
        try {
            rt.exec("killall charles")
            System.out.println("Stopping Charles ")
        } catch (e: IOException) {

        }

    }

    fun converttoCSV() {
        try {
           rt.exec("charles convert session.chls session.csv")
            Thread.sleep(2000)
        } catch (e: IOException) {
        } catch (e: InterruptedException) {
        }

    }

    fun ConverttoCSV(Fromfilename: String, Tofilname: String) {
        try {
           rt.exec("charles convert " + files.getCanonicalFile() + File.separator + Fromfilename + " " + files.getCanonicalFile() + File.separator + Tofilname + "")
            Thread.sleep(3000)
            System.out.println("Converting Charles Session")
        } catch (e: IOException) {
        } catch (e: InterruptedException) {
        }

    }

    @Throws(InterruptedException::class)
    fun downloadCharlesSession() {
        try {
            rt.exec("curl -o session.chls -x http://localhost:8888 http://control.charles/session/download")
            Thread.sleep(3000)
        } catch (e: IOException) {
        }

    }


    @Throws(InterruptedException::class)
    fun DownloadCharlesSession(filename: String) {
        try {

             rt.exec("curl -o " + files.getCanonicalFile() + File.separator + filename + " -x http://localhost:8888 http://control.charles/session/download")
            System.out.println("Downloading Charles Session")
            Thread.sleep(3000)
        } catch (e: IOException) {
        }

    }

    fun startcharlesSession() {
        try {
           rt.exec("curl -v -x http://localhost:8888 http://control.charles/recording/start")
            System.out.println("Starting Charles Session")
            Thread.sleep(2000)
        } catch (e: Exception) {
        }

    }

    fun stopCharlesSession() {
        try {
            //Process p5 = Runtime.getRuntime().exec(charlesStop);
           rt.exec("curl -v -x http://localhost:8888 http://control.charles/recording/stop")
            System.out.println("Stopping Charles Session")
            Thread.sleep(2000)
        } catch (e: Exception) {
        }

    }




}